doc/sphinx/how-to/how-to-test-mir-for-a-release.md

```diff
@@ -0,0 +1,181 @@
+
+5. Check that the virtual platform can run and you can connect to it via a VNC:
+   ```sh
+   WAYLAND_DISPLAY=wayland-1 miral-app --platform-display-lib=mir:virtual \
+       --virtual-output=1280x1024 \
+       --add-wayland-extension=zwp_virtual_keyboard_manager_v1:zwlr_virtual_pointer_manager_v1:zwlr_screencopy_manager_v1
+   ```
+   After, in a separate VT, connect to the VNC:
+   ```sh
+   WAYLAND_DISPLAY=wayland-1 ubuntu-frame-vnc
+   ```
+   Then, check that you can connect via a VNC viewer:
+   ```sh
+   gvncviewer localhost
+   ```
+
+### Applications
+For each empty box in the matrix above, ensure that the following applications can start
+
+1. Test that the following QT Wayland applications can be started and can receive input:
+    ```sh
+    sudo apt install qtwayland5 kate qterminal
+    
+    # First...
+    kate
+   
+    # Then...
+    qterminal
+    ```
+2. Test that `weston-terminal` can be started and can receive input:
+    ```
+    sudo apt install weston
+    weston
+    ```
+3. Test that `glmark2-wayland` can be run:
+    ```sh
+    sudo apt install glmark2-wayland
+    glmark2-wayland
+    ```
+4. (If using gbm-kms on a system with multiple GPUs) Test hybrid support with `glmark2-wayland`
+    ```sh
+    sudo apt install glmark2-wayland
+    glmark2-wayland
+    DRI_PRIME=0 glmark2-wayland
+    DRI_PRIME=1 glmark2-wayland
+    ```
+    (If more than 2 GPUs, may do `DRI_PRIME=2 glmark2-wayland`, etc)
+5. Test that `gnome-terminal` can be started and can receive input:
+   ```sh
+   sudo apt install gnome-terminal
+   gnome-terminal
+   ```
+
+### Mir Console Providers
+
+Run with different console providers and ensure that the compositor can start:
+
+```sh
+miral-app --console-provider=vt --vt=4
+```
+- This requires running with root privileges 
+- You need to ensure that `XDG_RUNTIME_DIR` is set in the environment. If using `sudo`, 
+    it might strip this out; running something like `sudo env XDG_RUNTIME_DIR=/run/user/1000 miral-shell ...`
+    will ensure this is set. 
+
+```sh
+miral-app --console-provider=logind
+```
+- You can switch to vt4 and sign in
+
+```sh
+miral-app --console-provider=minimal 
+```
+- This is used when all others fail
+- This does not provide VT switching capabilities (Ctrl-Alt-F1, etc) 
+- This is _only_ used for the `gbm-x11`, `gbm-wayland`, and `virtual` platforms 
+
+## Window Manager Examples
+Run with different window managers and confirm that the window management
+styles work as expected:
+
+```sh
+miral-app --window-manager=floating # traditional floating window manager
+miral-app --window-manager=tiling # a tiling window manager
+miral-app -kiosk # a typical kiosk
+```
+
+### Run Compositors that are Built for Testing
+1. Run the `mir-test-smoke-runner` and confirm that the final output is:
+   `Smoke testing complete with returncode 0`.
+2. Run the tests found in `mir-test-tools` and confirm the contents on the screen. A useful script for doing this is:
```

Comment:
Testing the `mir-test-tools` snap seems different in character to the what precedes it and probably belongs in a "testing the downstream snaps" section, not with the "smoke tests".